profiling/src/profile/mod.rs

```diff
@@ -81,6 +83,42 @@ struct Sample {
     local_root_span_id_label_offset: Option<usize>,
 }
 
+struct UpscalingRule {
+    values_offset: Vec<usize>,
+    label_name: i64,
+    label_value: i64,
+    upscaling_info: UpscalingInfo,
+}
+
+impl UpscalingRule {
+    pub fn compute_scale(&self, values: &[i64]) -> f64 {
+        match self.upscaling_info {
+            UpscalingInfo::Poisson {
+                x,
+                y,
+                sampling_distance,
+            } => {
+                if values[y] == 0 {
+                    return 0_f64;
+                }
+
+                let avg = (values[x] / values[y]) as f64;
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏ What happens if `avg==0` (eg. when `value[x]==0`) ?

Reply:
there will computation with this statement
`1_f64 / (1_f64 - (-0 / sampling_distance as f64).exp())` replacing avg with 0
Which might end up with to `1_f64`

Reply:
More like `+Inf` ;)

Reply:
oh I forgot the `exp()` :)

Reply:
I missed the `exp()`.
In that case, what do you propose to return ?

Reply:
That's what I had in mind, good point
When adding the rule, we can refuse the rule if the `distance_sampling` is 0. WDYT?

Reply:
While I agree that `value[x]` should not be 0 if `value[y]` is not 0, nothing prevents you from adding samples with 0 values. In the case `value[x]` or `value[y]` is 0, I would just return `1` (or `0`, but `1` is the neutral value for scaling).